pkg/operation/botanist/component/kubeapiserverexposure/kube_apiserver_service.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeapiserverexposure
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/kubeapiserver"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	servicePortName = "kube-apiserver"
+	servicePort     = 443
```

Comment:
I was referring to this which seems to be the same as `kubepapiserver.Port`. You can argue that the `port` and `targetPort` could differ, but it does not seem to be the case here. Therefore, I would simply reuse `kubeapiserver.Port` here. Anyway, this is just nitpicking. No hard feelings if you think it is better otherwise.

Reply:
> You can argue that the port and targetPort could differ

Exactly, that was exactly the reason why I didn't do it :) But you're right, most likely their values will always be the same.
Still, let's keep it for now 👍🏻 I might clean it up when it comes to the refactoring of the `kube-apiserver` Helm chart.